mpags-cipher.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main(int argc, char* argv[])
+{
+    const std::vector<std::string> cmdLineArgs { argv, argv+argc };
+    const size_t n = cmdLineArgs.size();
+
+    for (std::size_t i{0}; i < n; i++) {
+        // print help message and version information
+        if (cmdLineArgs[i] == "-h" || cmdLineArgs[i] == "--help") {
+            std::cout << "some help messages" << std::endl;
+        }
+
+        else if (cmdLineArgs[i] == "--version") {
+            std::cout << "version 0.1.0" << std::endl;
+        }
+
+        // check input and output file
+        else if (cmdLineArgs[i] == "-i") {
+            if (i != (n-1) && cmdLineArgs[i+1][0] != '-') {
+                std::cout << "input file is " << cmdLineArgs[i+1] << std::endl;
+            }
+            else std::cout << "declare input file name" << std::endl;
+        }
```

Comment:
Good that you check whether this is the last argument before trying access the next one.
However, it seems unnecessary to disallow filenames that start with '-', even if that is rather unlikely.
You also need to increment `i` so that the loop continues with the argument after the filename.
Error messages should conventionally be written to `std::cerr` instead of `std::cout`.